buildroot-external/rootfs-overlay/usr/sbin/hassos-supervisor

```diff
@@ -9,16 +9,72 @@ set -e
 . /etc/os-release
 
 # Init supervisor
+SUPERVISOR_MIN_RUNTIME=30
 SUPERVISOR_IMAGE="homeassistant/${SUPERVISOR_ARCH}-hassio-supervisor"
 SUPERVISOR_DATA=/mnt/data/supervisor
-SUPERVISOR_IMAGE_ID=$(docker inspect --format='{{.Id}}' "${SUPERVISOR_IMAGE}")
+SUPERVISOR_IMAGE_ID=$(docker inspect --format='{{.Id}}' "${SUPERVISOR_IMAGE}" || echo "")
 SUPERVISOR_CONTAINER_ID=$(docker inspect --format='{{.Image}}' hassio_supervisor || echo "")
 
-runSupervisor() {
+# Check if previous runtime of the container meets the minimal runtime
+# If not, we might be in trouble. Image or container corruption, bad release?
+# Delete the container, delete the image, pull a fresh one
+if [ -n "${SUPERVISOR_CONTAINER_ID}" ]; then
+    SUPERVISOR_START=$(docker inspect --format='{{.State.StartedAt}}' hassio_supervisor | sed -re "s/([-0-9]+)T([0-9\:]+).*/\1 \2/g")
+    SUPERVISOR_STOP=$(docker inspect --format='{{.State.FinishedAt}}' hassio_supervisor | sed -re "s/([-0-9]+)T([0-9\:]+).*/\1 \2/g")
+
+    if [ -n "${SUPERVISOR_START}" ] && [ -n "${SUPERVISOR_STOP}" ]; then
+        START=$(date --date="${SUPERVISOR_START}" +%s)
+        STOP=$(date --date="${SUPERVISOR_STOP}" +%s)
+        SUPERVISOR_RUNTIME=$((${STOP}-${START}))
+        # Minimial runtime not met, remove container and image
+        if [ "${SUPERVISOR_RUNTIME}" -lt "${SUPERVISOR_MIN_RUNTIME}" ]; then
+            echo "[WARNING] Supervisor container exited too quickly, forcing a fresh image and container..."
+            docker container rm --force hassio_supervisor || true
+            SUPERVISOR_CONTAINER_ID=""
+            docker rmi --force "${SUPERVISOR_IMAGE_ID}" || true
+            SUPERVISOR_IMAGE_ID=""
+        fi
+    fi
+fi
+
+# If Supervisor image is missing, pull it
+if [ -z "${SUPERVISOR_IMAGE_ID}" ]; then
+    # Get the latest from update information
+    # Using updater information instead of config. If the config version is
+    # broken, this creates a way (e.g., bad release).
+    SUPERVISOR_VERSION_UPDATER=$(jq -r '.version // empty' "${SUPERVISOR_DATA}/updater.json" || echo "")
+    SUPERVISOR_VERSION="${SUPERVISOR_VERSION_UPDATER:-latest}"
+
+    echo "[WARNING] Supervisor image missing, downloading a fresh one: ${SUPERVISOR_VERSION}"
```

Comment:
Why is this a warning? Didn't we just delete it on purpose?

Reply:
It doesn't have to be deleted on purpose by the system. This block stands for its own and has no relation to the if block above.

e.g., as a user, you can prune or delete your Supervisor Docker image